doc/icache.rst

```diff
@@ -149,3 +167,53 @@ Cache invalidation
 After reset, and when requested by the core (due to a FENCE.I instruction), the whole cache is invalidated.
 Requests are inserted to invalidate the tag RAM for all ways in each cache line in sequence.
 While the invalidation is in-progress, lookups and instruction fetches can proceed, but nothing will be allocated to the cache.
+
+Detailed behaviour
+^^^^^^^^^^^^^^^^^^
+
+This section describes the expected behaviour of the cache, in order to allow functional verification.
+This isn't an attempt to describe the cache's performance characteristics.
+
+The I$ has a single clock (``clk_i``) and asynchronous reset (``rst_ni``).
+
+Data is requested from the instruction memory with the ports prefixed by ``instr_``. These work as described in :ref:`instruction-fetch`.
+Note that there's one extra port on the I$, which doesn't appear at the ``ibex_core`` top-level.
+This is ``instr_pmp_err_i``.
+If the PMP block disallows a fetch for a certain address, it will squash the outgoing memory request entirely and set ``instr_pmp_err_i``.
+If that happens, the cache behaves as if the request was granted but returned some time later with an error.
+Note that it is possible for ``instr_gnt_i`` and ``instr_pmp_err_i`` to be high on the same cycle (if the memory system ties ``instr_gnt_i`` to ``instr_req_o``, for example).
```

Comment:
This is kind-of the other way around.
It's possible `instr_gnt_i` will never go high on a PMP error if the thing connected downstream ties `instr_gnt_i` to `instr_req_o`. On the other hand, if `instr_gnt_i` is wired up more like a conventional ready signal, it's possible that `instr_gnt_i` and `instr_pmp_err_i` would be high at the same time.